### ISSUE TITLE
Build System(s) are a Mess

### ISSUE THREAD

**Issue by blumf** [mariuz/flamerobin#120]
Looking at ditching the Boost dependency, and the code part is easy, but changing the various build systems we have is a real PITA.

Theoretically, we just use the bake file, but that it out of date, both in format (current Bakefile uses a different file format, legacy doesn't support recent IDEs) and content (the `flamerobin.bkl` file is missing some recent source files)

So, rather than trying to repair that, does anyone want to look at my [CMakeLists.txt](https://github.com/mariuz/flamerobin/files/5292751/CMakeLists.txt) file?

I've tested it on Windows (VS2019) and Linux (Slackware 14.2). It calls the `update-revision-info` script, builds, and sets up the `*-templates` directories ready for debugging. All that's missing is the install commands (for Linux, i.e. `make install`)

Special feature, for Windows devs, is you can directly specify the wxWidgets root (WXDIR) which makes it easier to try different wx versions on the same machine (it should check the WXDIR environment var too, but untested)

Quick guide for Win/Visual Studio use:

- Grab CMake from https://cmake.org/download/ (I recommend adding it to `PATH`)
- Place the [CMakeLists.txt](https://github.com/mariuz/flamerobin/files/5292751/CMakeLists.txt) in the root of the FlameRobin  (where all the other project file currently exist)
- Create a sub-dir 'build' (can be any name, like 'build-wx302-64bit', but 'build' for now)
- Run the CMake GUI
- Point 'Where is the source code' to the FlameRobin root
- Point 'Where to build the binaries' to the 'build' sub-dir
- Click 'Add Entry' (button to the right of window)
  - Give it the name WXDIR
  - Set Type to PATH
  - Select the root of your wxWidgets library
- Now click 'Generate'
- Select your Visual Studio version, and platform type (x86)
- If that all worked without issue, you can click 'Open Project' and you're into Visual Studio ready to build

I'm not the biggest fan of CMake, but it does seem to be the standard. Should we switch?

**Comment by mariuz** [comment]
I agree that cmake is the way to go , then we can add vcpkg depdendency for wxwidgets

**Comment by blumf** [comment]
Are people doing okay with the CMake system?

**Comment by arvanus** [comment]
I still couldn't test it, I had a busy week

**Comment by blumf** [comment]
Before I forget what I did, PR #129 changes the workflow test to use the new CMake system and drop the Boost requirements.

It's working, so we can look at the final step in all this, removing the old build files.

**Comment by blumf** [comment]
It's been a while (work), would everybody be okay if we get rid of the old build files now?
